source/Projectile.cpp

```diff
@@ -417,12 +419,12 @@ void Projectile::CheckLock(const Ship &target)
 			double rangeFraction = min(1., distance / jammingRange);
 			opticalJamming = (1. - rangeFraction) * opticalJamming;
 		}
-		double weight = target.Mass() * target.Mass();
-		double probability = weapon->OpticalTracking() * weight / (150000. + weight) / (1. + opticalJamming);
+		double weight = target.Mass() * target.Mass() / 1000000.;
+		double probability = (weapon->OpticalTracking() * weight / (1. + weight)) / (1. + opticalJamming);
```

Comment:
```suggestion
		double probability = weapon->OpticalTracking() * weight / (1. + weight) / (1. + opticalJamming);
```

Reply:
the two division symbols make me wanna simplify to just one division, as that'd halve the cost of this function (division is expensive)

Reply:
yeah that wasn't too hard, simple distributive property of multiplication
```suggestion
		double probability = weapon->OpticalTracking() * weight / ((1. + weight) * (1. + opticalJamming));
```
alternatively, there is this alternative implementation that makes optical jamming less effective on (excessively) larger vessels:
```suggestion
		double probability = weapon->OpticalTracking() * weight / ((1. + weight + opticalJamming));
```
For example, assuming 100% optical tracking, a ship of ~580t is tracked at 25% efficacy, 1000t is tracked with 50% efficacy, a ship of 2000t is tracked with 80% efficacy, and a ship of, say 5000t is tracked at 96.15% efficacy
At one optical jamming, i.e. halving the tracking effectiveness, the current proposal brings those numbers down to
580t: 25% to 12.5%
1000t: 50% to 25%
2000t: 80% to 40%
5000t: 96.15% to 48.07%

My proposal instead brings those numbers down non-linearly, and still has an asymptote at 100% efficacy at infinite tonnage:
580t: 25% to 14.4%
1000t: 50% to 33.33%
2000t: 80% to 66.67%
5000t: 96.15% to 92.59%

(Here is a graph showing same scale- Y axis is tracking %, X axis is ship tonnage, where the solid green line is the current proposal, `Optical Tracking * (weight / (1+weight)) * (1 / (1+jamming))`, and the dotted red line is my proposal, `Optical Tracking * weight / (1+weight+jamming)`, NOTABLY this first image is at jamming = 1, i.e. half effectiveness, and this second image is at jamming = 9, i.e. 10% effectiveness, showing how its behavior may be undesirable)
![image](https://github.com/user-attachments/assets/8075e629-1e5f-4290-9af3-a66aea853b46)
![image](https://github.com/user-attachments/assets/6eea7261-d6aa-4c7a-8d86-c3e5c64840d4)

Reply:
By the way, equations that produce these curves
![image](https://github.com/user-attachments/assets/1192549a-5f45-4865-bb03-654f757d83ea)

Reply:
I want to take a separate look at the mechanics of both optical and radar tracking in another PR, so I'll avoid making any mechanical changes to them in this PR, because they're not what this PR is about.